Python/bytecodes.c

```diff
@@ -2431,21 +2419,21 @@ dummy_func(
             if (Py_IsTrue(cond)) {
                 STACK_SHRINK(1);
                 _Py_DECREF_NO_DEALLOC(cond);
-                DISPATCH();
             }
-            if (Py_IsFalse(cond)) {
+            else if (Py_IsFalse(cond)) {
                 JUMPBY(oparg);
-                DISPATCH();
             }
-            err = PyObject_IsTrue(cond);
-            if (err > 0) {
-                STACK_SHRINK(1);
-                Py_DECREF(cond);
+            else {
+                err = PyObject_IsTrue(cond);
+                if (err > 0) {
+                    STACK_SHRINK(1);
+                    Py_DECREF(cond);
+                }
+                else if (err == 0)
+                    JUMPBY(oparg);
+                else
+                    goto error;
```

Comment:
Maybe it's time to add these braces? I shiver whenever I see an un-braced macro...

```suggestion
                else if (err == 0) {
                    JUMPBY(oparg);
                }
                else {
                    goto error;
                }
```